File: cpp/0704二分查找.cpp
#include<vector>

using namespace std;

int serch(vector<int> &nums,int target)
{
    int left = 0;
    int right = nums.size() - 1;
    int mid;
    while (left <= right)
    {
        mid = (left + right) >> 1;

        if(nums[mid]==target)
            return mid;
        
        if(nums[mid]<target)
            left = mid + 1;
        else
            right = mid - 1;
    }
    return -1;
}

File: python/0704二分查找.py
nums = [-1,0,3,5,9,12]
target = 9

def serch(nums:list,target:int):

    left=0
    right=len(nums)-1

    while(left<=right):
        mid=(left+right)>>1
        if(nums[mid]>target):
            right=mid-1
        elif(nums[mid]<target):
            left=mid+1
        else:
            return mid
    return -1

